test/functions_linearalgebra.jl

```diff
@@ -31,18 +31,20 @@ end
 @testset "svd" begin
     nda = NamedDimsArray{(:foo, :bar)}([1.0 2; 3 4])
     x = svd(nda)
+    # Test based on visualization on wikipedia
+    # https://en.wikipedia.org/wiki/File:Singular_value_decomposition_visualisation.svg
     @test dimnames(x.U) == (:foo, :_)
     @test dimnames(x.S) == (:_,)
```

Comment:
It might be neat if `S` could also have a name. This could easily be stored `NamedFactorization{(:foo, :bar, :svd)}`, and provided... maybe as an extra keyword, `svd(M; name=:svd)`?

I would even be tempted to make the default something other than `:_`. It could be the function name, `:svd`, or it could be something it's very unlikely people would type, like `:∑`.

Reply:
What would the benefit be though? You can figure out that it's an `SVD` from the factorization type it's parameterized by. I suppose it'd be nice if you could retain the original names when you do `Diagonal(x.S)`, but that seems like a weird special case.

Reply:
It would mean that `x.Vt * A` with some other `A` would retain this marker of where that dimension came from.

Reply:
Okay, but then in that case we'd need a different symbol than `:∑` for each case. I suppose we could insert the name of the function that would have inserted `:_` more generally, but that feels like it's solving a different problem.

Reply:
I had this debate extensively, though perhaps not publically, and perhaps only internally in my head when writing these originally.
Its out of scope for this PR.
Weird things can happen for any names you automatically provide.
Because each factorization executed is effectively pointing at a different latent space.
On that basis one could always gensym the names since it should *only* be allowed to by multiplied by its partner created at the same time .

Reply:
But lets not get into it in this PR.
It can be a follow up

Reply:
Right, that's the argument agains a fixed default name other than `:_`. (Perhaps instead of gensym it could be generated from the existing names.)

But without settling that, the ability to store a name which the caller provides seems easy to do in this approach. And valuable for the same reason that attaching names to anything is valuable. I guess it's an argument in favour of this struct, vs. the `getproperty` trick which lacks anywhere to store things.

Reply:
right, in FFT we gave ability to pass in names didn't we?
How would it look if we copies that API here?

Reply:
For FFT I never finished the PR. But instead of `fft(A, 2)` from FFTW, it was going to allow `fft(nA, :y => :k)` to specify the new name, and for `fft(nA, :y)` generate `:y∿` (or the reverse). That's simpler, though, as the whole transformation is per-dimension.